File: src/main/scala/codechicken/multipart/handler/MultipartMod.scala
package codechicken.multipart.handler

import cpw.mods.fml.common.Mod
import cpw.mods.fml.common.event.{
  FMLInitializationEvent,
  FMLPostInitializationEvent,
  FMLPreInitializationEvent,
  FMLServerAboutToStartEvent,
  FMLServerStoppedEvent
}
import cpw.mods.fml.common.Mod.EventHandler
import codechicken.multipart.MultiPartRegistry
import codechicken.multipart.Tags

@Mod(
  modid = "ForgeMultipart",
  name = "Forge Multipart",
  acceptedMinecraftVersions = "[1.7.10]",
  version = Tags.VERSION,
  modLanguage = "scala"
)
object MultipartMod {
  @EventHandler
  def preInit(event: FMLPreInitializationEvent) {
    MultipartProxy.preInit(event.getModConfigurationDirectory, event.getModLog)
  }

  @EventHandler
  def init(event: FMLInitializationEvent) {
    MultipartProxy.init()
  }

  @EventHandler
  def postInit(event: FMLPostInitializationEvent) {
    if (MultiPartRegistry.required) {
      MultiPartRegistry.postInit()
      MultipartProxy.postInit()
    }
  }

  @EventHandler
  def beforeServerStart(event: FMLServerAboutToStartEvent) {
    MultiPartRegistry.beforeServerStart()
  }

  @EventHandler
  def serverStopped(event: FMLServerStoppedEvent) {
    MultipartSaveLoad.loadingWorld = null;
  }
}

File: src/main/scala/codechicken/multipart/handler/MultipartMod.java
package codechicken.multipart.handler;

import codechicken.multipart.Tags;
import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.event.FMLInitializationEvent;
import cpw.mods.fml.common.event.FMLPostInitializationEvent;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import cpw.mods.fml.common.event.FMLServerAboutToStartEvent;
import cpw.mods.fml.common.event.FMLServerStoppedEvent;

@Mod(
        modid = "ForgeMultipart",
        name = "Forge Multipart",
        acceptedMinecraftVersions = "[1.7.10]",
        version = Tags.VERSION,
        modLanguage = "scala")
public final class MultipartMod {

    private MultipartMod() {}

    @Mod.EventHandler
    public static void preInit(FMLPreInitializationEvent event) {
        MultipartMod$.MODULE$.preInit(event);
    }

    @Mod.EventHandler
    public static void init(FMLInitializationEvent event) {
        MultipartMod$.MODULE$.init(event);
    }

    @Mod.EventHandler
    public static void postInit(FMLPostInitializationEvent event) {
        MultipartMod$.MODULE$.postInit(event);
    }

    @Mod.EventHandler
    public static void beforeServerStart(FMLServerAboutToStartEvent event) {
        MultipartMod$.MODULE$.beforeServerStart(event);
    }

    @Mod.EventHandler
    public static void serverStopped(FMLServerStoppedEvent event) {
        MultipartMod$.MODULE$.serverStopped(event);
    }
}
